prov/usnic/src/usdf_dgram.c

```diff
@@ -531,17 +621,72 @@ usdf_dgram_prefix_sendv(struct fid_ep *fep, const struct iovec *iov, void **desc
 		send_iov[0].iov_base = ((char *) send_iov[0].iov_base +
 				USDF_HDR_BUF_ENTRY);
 		send_iov[0].iov_len -= USDF_HDR_BUF_ENTRY;
-		_usdf_dgram_send_iov_copy(ep, dest, send_iov, count, context);
+
+		return _usdf_dgram_send_iov_copy(ep, dest, send_iov, count,
+				context, ep->ep_tx_completion);
+	} else if (ep->e.dg.tx_op_flags & FI_INJECT) {
+		USDF_DBG_SYS(EP_DATA,
+				"given inject length (%zu) exceeds max inject length (%d)\n",
+				len, USD_SEND_MAX_COPY);
+		return -FI_ENOSPC;
```

Comment:
I don't think it matters too much.  I usually think of `ENOSPC` as indicating a lack of _disk_ space, but I'm sure I've probably used it to have your meaning in code that I've written in the past.  `ENOBUFS` is another choice here, but it's all bike-shedding, so we should probably just leave it as `ENOSPC` in your PR.